code/modules/surgery/bodyparts/dismemberment.dm

```diff
@@ -277,6 +277,10 @@
 
 /obj/item/bodypart/proc/attach_limb(mob/living/carbon/C, special)
 	moveToNullspace()
+	if(SEND_SIGNAL(C, COMSIG_LIVING_ATTACH_LIMB, src, special) & COMPONENT_NO_ATTACH)
+		forceMove(C.loc)
```

Comment:
You should instead just prevent the move and let the caller handle where it should go.